src/UglyToad.PdfPig/Writer/TokenWriter.cs

```diff
@@ -407,40 +408,70 @@ private static void WriteStream(StreamToken streamToken, Stream outputStream)
             WriteLineBreak(outputStream);
             outputStream.Write(StreamEnd, 0, StreamEnd.Length);
         }
-
+
+        private static int[] EscapeNeeded = new int[]
+        {
+            '\r', '\n', '\t', '\b', '\f', '\\'
+        };
+        private static int[] Escaped = new int[]
+        {
+            'r', 'n', 't', 'b', 'f', '\\'
+        };
         private static void WriteString(StringToken stringToken, Stream outputStream)
-        {
+        {
+            outputStream.WriteByte(StringStart);
+
             if (stringToken.EncodedWith == StringToken.Encoding.Iso88591)
-            {
-                var isUtf16 = false;
-                for (var i = 0; i < stringToken.Data.Length; i++)
-                {
-                    var c = stringToken.Data[i];
-
-                    if (c == (char) StringStart || c == (char)StringEnd || c == (char) Backslash)
-                    {
-                        stringToken = new StringToken(stringToken.Data.Insert(i++, "\\"), stringToken.EncodedWith);
-                    }
-
-                    // Close enough.
-                    if (c > 250)
-                    {
-                        isUtf16 = true;
-                        break;
-                    }
-                }
-
-                if (isUtf16)
-                {
-                    stringToken = new StringToken(stringToken.Data, StringToken.Encoding.Utf16BE);
-                }
-            }
-
-            outputStream.WriteByte(StringStart);
-            var bytes = stringToken.GetBytes();
-            outputStream.Write(bytes, 0, bytes.Length);
-            outputStream.WriteByte(StringEnd);
-
+            {
+                // iso 88591 (or really PdfDocEncoding in non-contentstream circumstances shouldn't
+                // have these chars but seems like internally this isn't obeyed (see:
+                // CanCreateDocumentInformationDictionaryWithNonAsciiCharacters test) and it may
+                // happen during parsing as well -> switch to unicode
+                if (stringToken.Data.Any(x => x > 255))
+                {
+                    var data = new StringToken(stringToken.Data, StringToken.Encoding.Utf16BE).GetBytes();
+                    outputStream.Write(data, 0, data.Length);
+                }
+                else
+                {
+                    int ei;
+                    for (var i = 0; i < stringToken.Data.Length; i++)

```

Comment:
did you consider putting this code in the `GetBytes()` method of the `StringToken` class, under the `Iso88591` case in the switch on encodindWith ? I feel like that could be a bug, but I could be wrong.

Reply:
I did but looked like other places were using the GetBytes() method to get the raw byte data that was encoded in the string (not serialized byte output)